src/main/kotlin/com/autonomousapps/model/declaration/Configurations.kt

```diff
@@ -10,7 +10,8 @@ internal object Configurations {
   internal const val CONF_ADVICE_ALL_CONSUMER = "adviceAllConsumer"
   internal const val CONF_ADVICE_ALL_PRODUCER = "adviceAllProducer"
 
-  private val MAIN_SUFFIXES = listOf("api", "implementation", "compileOnly", "runtimeOnly")
+  private val COMPILE_ONLY_SUFFIXES = listOf("compileOnly", "compileOnlyApi", "providedCompile")
```

Comment:
I think this is fine in practice, but in the interest of thinking aloud, it seems unlikely that Android would ever support something like `debugProvidedCompile`. And of course android-land is the reason we have to deal with things like suffixes and prefixes...